pkg/metrics/githubapp.go

```diff
@@ -0,0 +1,95 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"github.com/bradleyfalzon/ghinstallation/v2"
+	"github.com/google/go-github/v45/github"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/redhat-appstudio/application-service/gitops"
+	gitopsprepare "github.com/redhat-appstudio/application-service/gitops/prepare"
+	"github.com/redhat-appstudio/build-service/pkg/boerrors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"net/http"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+	"strconv"
+)
+
+type GithubAppAvailabilityProbe struct {
+	client                  client.Client
+	gauge                   prometheus.Gauge
+	getGithubAppCredentials func(ctx context.Context, client client.Client) (int64, []byte, error)
+	getGithubApp            func(ctx context.Context, tr http.RoundTripper, appID int64, privateKey []byte) (*github.App, *github.Response, error)
+}
+
+func NewGithubAppAvailabilityProbe(client client.Client) *GithubAppAvailabilityProbe {
+	return &GithubAppAvailabilityProbe{
+		client: client,
+		gauge: prometheus.NewGauge(
+			prometheus.GaugeOpts{
+				Namespace: MetricsNamespace,
+				Subsystem: MetricsSubsystem,
+				Name:      "global_github_app_available",
+				Help:      "The availability of the Github App",
+			}),
+		getGithubAppCredentials: githubAppCredentials,
+		getGithubApp:            getGithubApp,
+	}
+}
+
+func (g *GithubAppAvailabilityProbe) CheckAvailability(ctx context.Context) error {
+
+	githubAppId, privateKey, err := g.getGithubAppCredentials(ctx, g.client)
+	if err != nil {
+		return err
+	}
+	_, _, err = g.getGithubApp(ctx, http.DefaultTransport, githubAppId, privateKey)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func githubAppCredentials(ctx context.Context, client client.Client) (int64, []byte, error) {
+	log := ctrllog.FromContext(ctx).V(1)
+	pacSecret := corev1.Secret{}
+	globalPaCSecretKey := types.NamespacedName{Namespace: BuildServiceNamespaceName, Name: gitopsprepare.PipelinesAsCodeSecretName}
+	log.Info("Checking GitHub App availability")
+	if err := client.Get(ctx, globalPaCSecretKey, &pacSecret); err != nil {
+		return 0, nil, boerrors.NewBuildOpError(boerrors.EPaCSecretNotFound,
+			fmt.Errorf(" Pipelines as Code secret not found in %s namespace nor in %s", BuildServiceNamespaceName, globalPaCSecretKey.Namespace))
```

Comment:
The error message here seems to say that we looked in two different namespace, but didn't we look only in the `BuildServiceNamespace`?